examples/intRVFL.py

```diff
@@ -0,0 +1,539 @@
+import torch
+import torch.nn as nn
+from tqdm import tqdm
+
+# Note: this example requires the torchmetrics library: https://torchmetrics.readthedocs.io
+import torchmetrics
+
+import torchhd
+from torchhd import embeddings
+from torchhd.map import MAP
+
+
+# Function that performs the encoding according to intRVFL model. See details in `Density Encoding Enables Resource-Efficient Randomly Connected Neural Networks <https://doi.org/10.1109/TNNLS.2020.3015971>`_.
+def encoding_density_clipped(
+    keys,
+    values,
+    kappa: int,
+):
+    # Perform binding of key and value vectors
+    sample_hv = MAP.bind(keys, values)
+    # Perform the superposition operation on the bound key-value pairs
+    sample_hv = MAP.multibundle(sample_hv)
+    # DK: consider making this function into the functinal.py - could be useful in other contexts
+    # Perform clipping function on the result of the superposition operation
+    sample_hv[sample_hv > kappa] = kappa
+    sample_hv[sample_hv < -kappa] = -kappa
+    return sample_hv
+
+
+# Function that forms the classifier (readout matrix) with the ridge regression
+def classifier_ridge(
+    train_ld: torch.utils.data.dataloader.DataLoader,
+    min_val: torch.float,
+    max_val: torch.float,
+    dimensions: int,
+    num_classes: int,
+    lamb: float,
+    encoding_function,
+    data_type: torch.dtype,
+    device: torch.device,
+):
+
+    # Get number of training samples
+    num_train = train_ld.dataset.targets.size(0)
+    # Collects high-dimensional represetations of data in the train data
+    total_samples_hv = torch.zeros(
+        num_train,
+        dimensions,
+        dtype=data_type,
+        device=device,
+    )
+    # Collects one-hot encodings of class labels
+    labels_one_hot = torch.zeros(
+        num_train,
+        num_classes,
+        dtype=data_type,
+        device=device,
+    )
+
+    with torch.no_grad():
+        count = 0
+        for samples, labels in tqdm(train_ld, desc="Training"):
+
+            samples = samples.to(device)
+            labels = labels.to(device)
+            # Make one-hot encoding
+            labels_one_hot[count, labels] = 1
+
+            # DK: I think this needs to be somehow connected to transform (callable, optional) that we have for datasets
+            # Normalize
+            samples = (samples - min_val) / (max_val - min_val)
```

Comment:
Yes, using the transform calls would be great but we need to think about at which stage you would get the min, max values of each dataset.

Reply:
Currently, decided to get them as a part of "evaluate(self)" but still not entirely happy about this